corehq/apps/reports/forms.py

```diff
@@ -313,6 +326,21 @@ def __init__(self, data, *args, **kwargs):
             )
         )
 
+        if self.add_allowed_tableau_groups_field:
+            self.helper.layout.insert(
+                -1,
+                'tableau_groups_allowed',
+            )
+
+    def _setup_tableau_groups_allowed_field(self, kwargs, user_syncing_config):
+        self.all_tableau_groups = user_syncing_config['all_tableau_groups']
```

Comment:
Can we define `self.all_tableau_groups` in the init, and avoid needing a `add_allowed_tableau_groups_field` flag? 

In the save function, the check can be `if self.all_tableau_groups`

Reply:
Good point, I think it did help me see the relation.

In that case, I think we can define `self.add_allowed_tableau_groups_field = user_syncing_config['user_syncing_enabled'] and user_syncing_config.get('server_reachable')` first instead of within the conditional?